### ISSUE TITLE
[Sharing Experience] Training Z-Image LoRA using 12G VRAM ~ 😁

### ISSUE THREAD

**Issue by juntaosun** [ostris/ai-toolkit#550]
(1) Datasets Training Preparation: Image material with a maximum side size of 768.

🎉 To minimize VRAM usage, after extensive testing, training was successful with 6~10 images, yielding good LoRa results.  
>  A maximum image side resolution of 1024 might cause insufficient 12G VRAM. You can try it!  

(2) New Job Creation: Select Z-Image Turbo, and then set your model path. Follow the settings in the screenshot below.

<img width="1315" height="672" alt="Image" src="https://github.com/user-attachments/assets/f68ddc8e-b797-49cf-b60a-8f8e7a22e972" />

You need to enter your own trigger words! This is just an example !  
⚠️ Remember to set the Transformer Offload to 0%; we won't use it because it will throw an error, and we're unsure if this is a bug.  

<img width="1668" height="351" alt="Image" src="https://github.com/user-attachments/assets/3fd4cd1a-91cb-4c92-892d-3901c36385af" />

Correction: Learning Rate 0.0001 ~ 0.0002 !  
👉 After extensive testing, 2000 steps is a suitable value. 

<img width="1522" height="391" alt="Image" src="https://github.com/user-attachments/assets/570e7bf2-74f7-4611-9c7f-4fda0f256c49" />

Datasets: Cache Latents and Resolution 512, 768 ~
>  A maximum image side resolution of 1024 might cause insufficient 12G VRAM. You can try it!  

<img width="693" height="228" alt="Image" src="https://github.com/user-attachments/assets/c0870a04-8a76-45f4-9f39-ea9be49660f9" />

<img width="568" height="222" alt="Image" src="https://github.com/user-attachments/assets/7a236473-dca4-4699-a274-2f1541759862" />

👆 As you can see, training started successfully on 12G VRAM, and the speed is quite good !👇

```
first_lora_v1:  26%|##5       | 519/2000 [24:43<1:10:54,  2.87s/it, lr: 2.0e-04 loss: 3.811e-01]
```
Training speed is approximately 2~3 seconds/it, 2000 steps take about 1 - 2 hour to complete.

Finally: Wishing users with low VRAM success in training their own z-image LoRA!  
Thanks to ai-toolkit and z-image, have fun!  If you have better training settings, please share! 🤗

<img width="331" height="303" alt="Image" src="https://github.com/user-attachments/assets/b6b6c3f9-025b-46db-a4ea-9b6b96554f68" />

> This is my second z-image LoRA. ai-toolkit\output  

Now, you can use it in ComfyUI via the Lora loader !   Z-Image-Turbo ~

**Comment by leetraman822** [comment]
thank you, very useful.

**Comment by bank010** [comment]
How to train the checkpoints model?

**Comment by juntaosun** [comment]
@bank010 I trained Lora using Z-Image-Turbo.

**Comment by bank010** [comment]
> [@bank010](https://github.com/bank010) I trained Lora using Z-Image-Turbo.

您知道怎么微调Z-image-Turbo的大模型吗？不是lora微调，全量微调

**Comment by yamasoo** [comment]
5070TI 16G first try

<img width="3348" height="737" alt="Image" src="https://github.com/user-attachments/assets/063b384d-2423-4d7f-9805-f0634d716965" />

**Comment by juntaosun** [comment]
@yamasoo The 5070TI 16G can train at 1024 resolution.

**Comment by cmyknao** [comment]
@juntaosun Thank you for sharing your information. This was very usefull. After some testing I got 1024 resolution running on my RTX 3060 with 12 GB VRAM. I used the standard settings for Z-Image Turbo (like Transformer = float8 and Resolutions = 512, 768, 1024), exept the following changes:

1. Optimizer: Adafactor
2. Learning Rate = 0,0003
3. Steps = 1200
4. Cache Text Embeddings
5. Cache Latents
6. Sample Resolution: Width = 768, Hight = 1024 (with 1024 x 1024 it hangs just at the first sample)

It runs hard at the limit at about 11,5 - 11,7 GB VRAM usage. Speed was aprox. between 6 - 10 s/itteration. With Adafactor and a Learning Rate of 0.0003 approx. 1100 steps seems to be optimal.
I`m not really sure, but Cache Text Embeddings and Cache Latents seems to reduce the VRAM usage crucially.
Hope this helps some others too.

**Comment by juntaosun** [comment]
@cmyknao Thank you for sharing your information.

**Comment by wideload1971** [comment]
Thank you very much for these settings. I was trying to use another set that claimed to work on 12GB VRAM but kept getting an OOM near the start. It hasn't finished yet but the training is progressing, which is further than before. I suspect it may have been the cache latents or layer offloading that was the issue, as they were both off in my earlier attempts. Fingers crossed for this attempt.

**Comment by cmyknao** [comment]
@wideload1971 After starting, it uses more than 12 GB of VRAM for a short time. (Unfortunately, I can't remember at which step). But after that, everything runs smoothly.

**Comment by ladydarkness** [comment]
Thank you, works very well on my rig, 

Ryzen 9 3900 + RTX 4070 Super

**Comment by thinhlpg** [comment]
ty this is awesome :333

**Comment by 7ywx** [comment]
There is no z-image in the model architecture.

**Comment by Signorlimone** [comment]
> [@juntaosun](https://github.com/juntaosun) Thank you for sharing your information. This was very usefull. After some testing I got 1024 resolution running on my RTX 3060 with 12 GB VRAM. I used the standard settings for Z-Image Turbo (like Transformer = float8 and Resolutions = 512, 768, 1024), exept the following changes:
> 
>     1. Optimizer: Adafactor
> 
>     2. Learning Rate = 0,0003
> 
>     3. Steps = 1200
> 
>     4. Cache Text Embeddings
> 
>     5. Cache Latents
> 
>     6. Sample Resolution: Width = 768, Hight = 1024 (with 1024 x 1024 it hangs just at the first sample)
> 
> 
> It runs hard at the limit at about 11,5 - 11,7 GB VRAM usage. Speed was aprox. between 6 - 10 s/itteration. With Adafactor and a Learning Rate of 0.0003 approx. 1100 steps seems to be optimal. I`m not really sure, but Cache Text Embeddings and Cache Latents seems to reduce the VRAM usage crucially. Hope this helps some others too.

sorry in advance for the "tangential" question. Do you have to add the relative_step parameter to use Adafactor?

**Comment by wynandhuizinga** [comment]
@juntaosun Thanks for the elaborate instructions! (also @cmyknao !). I managed to train a lora on 1024 on my 3060 with 12gb. Trick for me was to leverage 15 percent Transformer offloading. Took me 7hours to train 3000 steps on a dataset of 46 images (1024x1024). Mind that I do have a machine with 48gb ram and a i5-10600k. 

Other tip I can share: turn off the testing images. Takes my machine 3 minutes to render, if you take 10 (default) sample images and create an intermittend lora every 250 steps. That means you'll be spending 30minutes * (3000/250) = 6hrs on for me pretty useless images. Those images are not utilized during the training process.

**Comment by ZAIGGENX** [comment]
That was very useful, Thanks for share

**Comment by Thanakit-KH** [comment]
The information I received from you has been very helpful. Thank you for sharing.

**Comment by cain666-4u** [comment]
thanks a bunch! im getting 1.5s/it with these settings with a 3080ti 12gb
